codeowners/src/github.rs

```diff
@@ -187,7 +185,54 @@ impl BindingsGitHubOwners {
     }
 }
 
-fn pattern(path: &str) -> anyhow::Result<Pattern> {
+#[derive(Debug, PartialEq, Clone, Eq)]
+pub struct PatternWithFallback {
+    base: Pattern,
+    fallback: Option<Pattern>,
+}
+
+impl PatternWithFallback {
+    pub fn new(base: &str) -> anyhow::Result<Self> {
+        // Matches anything that ends with neither a slash nor a period nor an asterisk,
+        // see test pattern_with_fallback for cases
+        let fallback_needed_regex = Regex::new(r"^\/?([a-zA-Z0-9_*\-]*\/)*([a-zA-Z0-9_\-]+)$")?;
```

Comment:
We should use `lazy_static` for this regex to avoid recompiling it on every invocation. Docs:
https://docs.rs/regex/latest/regex/#avoid-re-compiling-regexes-especially-in-a-loop